### PR TITLE
confirm deleting branch when it has an open PR

### LINKED ISSUE #3615: Unable to Delete local branches because of new "open pull request" dialog.
### Description

Unable to delete a local branch because a dialog box saying I have an open pull request for that branch. There is no open pull request.

### Version

**GitHub Desktop version:** 1.0.11
**OS version:** Windows 10

### Steps to Reproduce

1. Create a local branch. Make code changes. Push branch to origin.
2. In github, open a PR for the branch. A few minutes later, merge the PR into master.
3. In application, try to delete the branch (Branch / Delete...), get message saying I can't.

To "fix it", I need to switch to master, fetch from origin, then switch back to the branch and delete it.

This started with the latest release and is a pain now to have to switch back and forth.

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #3615 

 - [x] display a different dialog instead of blocking on a current open PR

### CODE REVIEW

**Comment by j-f1** [comment]
Does this dialog show up once the PR has been merged (or closed)?

**Comment by shiftkey** [comment]
> Does this dialog show up once the PR has been merged (or closed)?

We only track open pull requests in the app, but it might require a refresh in the app to see this change flow through. Hence wording things as a suggestion rather than something you must do.

File: app/src/ui/delete-branch/delete-pull-request-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      <Dialog
+        id="delete-branch"
+        title={__DARWIN__ ? 'Delete Branch' : 'Delete branch'}
+        type="warning"
+        onDismissed={this.props.onDismissed}
+      >
+        <DialogContent>
+          <p>This branch has an open pull request associated with it.</p>
+          <p>
+            If this pull request has been merged, you can also remove the remote
```
Comment (iAmWillShepherd) [inline app/src/ui/delete-branch/delete-pull-request-dialog.tsx]: The text here can be condensed a bit

```tsx
<p>
  If <LinkButton onClick={this.openPullRequest}>#{this.props.pullRequest.number}</LinkButton> has been merged, you can also remove the remote branch on GitHub.
</p>
```

**Comment by shiftkey** [comment]
:apple:

**Pull request merged**